common/src/test/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2StrategyTest.java

```diff
@@ -0,0 +1,126 @@
+
+@RunWith(MockitoJUnitRunner.class)
+public class OAuth2StrategyTest {
+
+    public class TestStrategy extends OAuth2Strategy {
+
+        @Override
+        protected HttpResponse performTokenRequest(TokenRequest request) throws IOException, ClientException {
+            return null;
+        }
+
+        /**
+         * Constructor of OAuth2Strategy.
+         *
+         * @param config             generic OAuth2 configuration
+         * @param strategyParameters
+         */
+        public TestStrategy(OAuth2Configuration config, OAuth2StrategyParameters strategyParameters) {
+            super(config, strategyParameters);
+        }
+
+        @Override
+        public AuthorizationResultFactory getAuthorizationResultFactory() {
+            return null;
+        }
+
+        @Override
+        public String getIssuerCacheIdentifier(AuthorizationRequest request) {
+            return null;
+        }
+
+        @Override
+        public AccessToken getAccessTokenFromResponse(TokenResponse response) {
+            return null;
+        }
+
+        @Override
+        public RefreshToken getRefreshTokenFromResponse(TokenResponse response) {
+            return null;
+        }
+
+        @Override
+        public BaseAccount createAccount(TokenResponse response) {
+            return null;
+        }
+
+        @Override
+        public AuthorizationRequest.Builder createAuthorizationRequestBuilder() {
+            return null;
+        }
+
+        @Override
+        public AuthorizationRequest.Builder createAuthorizationRequestBuilder(IAccountRecord account) {
+            return null;
+        }
+
+        @Override
+        public TokenRequest createTokenRequest(AuthorizationRequest request, AuthorizationResponse response, AbstractAuthenticationScheme authScheme) throws ClientException {
+            return null;
+        }
+
+        @Override
+        public TokenRequest createRefreshTokenRequest(AbstractAuthenticationScheme authScheme) throws ClientException {
+            return null;
+        }
+
+        @Override
+        protected void validateAuthorizationRequest(AuthorizationRequest request) {
+
+        }
+
+        @Override
+        protected void validateTokenRequest(TokenRequest request) {
+
+        }
+
+        @Override
+        protected TokenResult getTokenResultFromHttpResponse(HttpResponse response) throws ClientException {
+            //This will return a null TokenResponse.
+            return new TokenResult();
+        }
+
+        @Override
+        protected void validateTokenResponse(TokenRequest request, TokenResponse response) throws ClientException {
+
+        }
+    }
+
+    private static class TestParams extends OAuth2StrategyParameters {
+        @Nullable
+        @Override
+        public Context getContext() {
+            return null;
+        }
+    };
+
+    /**
+     * This test only verifies that if a null token response is returned from the token result,
+     * we don't create an error.
```

Comment:
don't create an error? So what is the expected outcome here?

Reply:
We don't dereference the null pointer, so therefore don't throw an exception.

Reply:
What do we expect to get back from the requestToken method in this case? If we are not going to throw exception, think we should assert for what we are getting back out of the requestToken method.

Reply:
Assert not null or assert that the same token result is returned that I created?

Reply:
Yes assertNotNull is fine